src/main/java/org/threadly/concurrent/future/ListenableFuture.java

```diff
@@ -61,7 +66,53 @@
    *          to invoke on this thread or future complete thread (depending on future state)
    * @return A new {@link ListenableFuture} with the specified result type
    */
-  public <R> ListenableFuture<R> map(Function<? super T, R> mapper, Executor executor);
+  public <R> ListenableFuture<R> map(Function<? super T, ? extends R> mapper, Executor executor);
+  
+  /**
+   * Similar to {@link #map(Function)}, in that this will apply a mapper function once the applied 
+   * to future completes.  Once this future resolves it will provide the result into the provided 
+   * function.  Unlike {@link #flatMap(Function)}, this will then unwrap a future provided from the 
+   * function so that instead of having {@code ListenableFuture<ListenableFuture<R>>} you can 
+   * simply extract the final value.  The returned future will only resolve once the future of the 
+   * provided function completes.  
+   * 
+   * If the future is complete already, the function may be invoked on the invoking thread.  If the 
+   * future is not complete then the function will be invoked on the thread which completes 
+   * the future (immediately after it completes).  
+   * 
+   * @since 5.0.0
+   * 
+   * @param <R> The type for the object contained in the future which is returned from the mapper
+   * @param mapper Function to invoke in order to transform the futures result
+   * @return A new {@link ListenableFuture} with the specified result type
+   */
+  public <R> ListenableFuture<R> flatMap(Function<? super T, ListenableFuture<R>> mapper);
+
+  /**
+   * Similar to {@link #map(Function, Executor)}, in that this will apply a mapper function once 
+   * the applied to future completes.  Once this future resolves it will provide the result into 
+   * the provided function.  Unlike {@link #flatMap(Function, Executor)}, this will then unwrap a 
```

Comment:
Yea, good suggestion, these probably deserve some code examples in the javadocs.  We really need to figure out a better way to handle example code (for everything really).  One guy was going to help the project with this, but he disappeared, haha.